beehive-lab/TornadoVM
# [refactor] GridTask renamed to GridScheduler

#### Description

This PR applies a change that we have discussed internally regarding the name of the `GridTask`. The fact that we can have multiple tasks that use the same `GridTask` object, has made us think that the naming is not appropriate as it makes developers think that one task corresponds to a `GridTask`.

Therefore we have agreed to rename the `GridTask` to `GridScheduler`. Here is an example of usage:
```java
        WorkerGrid worker = new WorkerGrid1D(size);
        GridScheduler gridScheduler = new GridScheduler();
        gridScheduler.setWorkerGrid("s02.t0", worker);
        gridScheduler.setWorkerGrid("s02.t1", worker);
        gridScheduler.setWorkerGrid("s02.t2", worker);
        KernelContext context = new KernelContext();

        TaskSchedule s0 = new TaskSchedule("s0") //
                .streamIn(a, b) //
                .task("t0", TestCombinedTaskSchedule::vectorAddV2, context, a, b, cTornado) //
                .task("t1", TestCombinedTaskSchedule::vectorMulV2, context, cTornado, b, cTornado) //
                .task("t2", TestCombinedTaskSchedule::vectorSubV2, context, cTornado, b, cTornado) //
                .streamOut(cTornado);
        s0.execute(gridScheduler);
```

No functional code has changed in this PR.

#### Backend/s tested


#### OS tested


#### Did you check on FPGAs?

If possible, check your changes on FPGAs.


#### How to test the new patch?

No need to test. For sanity check, you can use:
```bash
grep -nr "GridTask" .
```
